sagemaker-spark/parameterize-spark-config-pysparkprocessor-pipeline/parameterize-spark-config-pysparkprocessor-pipeline.ipynb

```diff
@@ -0,0 +1,518 @@
+    "    framework_version=\"3.1\",\n",
+    "    role=role,\n",
+    "    instance_count=2,\n",
+    "    instance_type=\"ml.m5.xlarge\",\n",
+    "    max_runtime_in_seconds=1200,\n",
+    "    sagemaker_session=sagemaker_session,\n",
+    ")\n",
+    "\n",
+    "step_args = pyspark_processor.run(\n",
+    "    \"./code/preprocess.py\",\n",
+    "    inputs=[\n",
+    "        ProcessingInput(\n",
+    "            source=spark_config_s3_uri,\n",
+    "            destination=f\"{pyspark_processor._conf_container_base_path}{pyspark_processor._conf_container_input_name}\",\n",
+    "            input_name=_SparkProcessorBase._conf_container_input_name,\n",
+    "        )\n",
+    "    ],\n",
+    "    arguments=[\n",
+    "        \"--s3_input_bucket\",\n",
+    "        default_bucket,\n",
+    "        \"--s3_input_key_prefix\",\n",
+    "        input_prefix_abalone,\n",
+    "        \"--s3_output_bucket\",\n",
+    "        default_bucket,\n",
+    "        \"--s3_output_key_prefix\",\n",
+    "        input_preprocessed_prefix_abalone,\n",
+    "    ],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Step-8: Create ProcessingStep and Pipeline \n",
+    "Create a processing step. This step takes in the PySparkProcessor, the input and output channels, and the ./code/preprocess.py script that we created. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.workflow.steps import ProcessingStep\n",
+    "from sagemaker.workflow.pipeline import Pipeline\n",
+    "\n",
+    "spark_step_process = ProcessingStep(name=\"AbaloneSparkProcess\", step_args=step_args)\n",
+    "\n",
+    "pipeline_name = f\"AbalonePipeline-Spark\"\n",
+    "pipeline = Pipeline(\n",
+    "    name=pipeline_name, parameters=[spark_config_s3_uri], steps=[spark_step_process]\n",
+    ")\n",
+    "\n",
+    "pipeline.upsert(role_arn=role)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we have successfully created a sagemaker pipeline with a PySparkProcessor. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Pipeline Executions\n",
+    "### Execute pipeline with default spark-configuration\n",
+    "If no SparkConfigS3Uri parameter value is provided, pipeline execution uses default_spark_configuration_object_s3_uri as a default spark-configuration. In the following execution example, we execute PySparkProcessor with default spark-configuration."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "execution_with_default_spark_configuration = pipeline.start()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can verify that PySparkProcessor is using the default spark-configuration by looking into the CloudWatch logs."
```

Comment:
This seems not a smooth experience, where should I find the right Processing Job to look at?

IMO, here we should add some instructions like `execution_with_default_spark_configuration.list_steps()` to locate the processing job first.

Reply:
> This seems not a smooth experience, where should I find the right Processing Job to look at?
> 
> IMO, here we should add some instructions like `execution_with_default_spark_configuration.list_steps()` to locate the processing job first.
> 
> Same for the next cell.
I added execution describe(), wait() and list_steps() as per suggestion in the new revision